service/matching/handler/engine.go

```diff
@@ -871,6 +871,10 @@ func (e *matchingEngineImpl) QueryWorkflow(
 	}
 
 	taskID := uuid.New()
+	queryResultCh := make(chan *queryResult, 1)
```

Comment:
nit: Can you modify the unit test for success case? Using below should be able to capture the race condition 

```
mockManager.EXPECT().DispatchQueryTask(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(ctx context.Context, taskID string, request *types.MatchingQueryWorkflowRequest) (*types.MatchingQueryWorkflowResponse, error) {
					queryResChan, ok := queryTaskMap.get(taskID)
					if !ok {
						return nil, errors.New("cannot find query result channel by taskID")
					}
					queryResChan <- &queryResult{workerResponse: &types.MatchingRespondQueryTaskCompletedRequest{
						// TO BE COMPELTED
					}}
					return nil, nil
				})
```